traitlets/config/tests/test_application.py

```diff
@@ -66,7 +66,7 @@ class MyApp(Application):
 
     aliases = Dict({
                     ('fooi', 'i') : 'Foo.i',
-                    ('j', 'fooj') : 'Foo.j',
+                    ('j', 'fooj') : ('Foo.j', "`j` terse help msg"),
```

Comment:
The same reasoning you explained could have been applied also in the flags, like that:
Since `.help` keys exist for all options a flag masp to, then just concatenate all of them in one big help-message, and use this for the flag - no need to have a shorter version in a 2-tuple.
But hopefully the case is not like that.

I have a need for this because my app has ~15 subcommands with 3 levels of nesting. And since i did a trick to *merge* aliases and flags from parent cmds, so at time the output of ` root foo bar  --help`  is simply overwhelming, with tens of of them.  So I'm using shorter versions to conserve screen estate and help the users see what's important.

I repeat this PR provides an optional functionality, so by default the longer-help message is used.